src/hera/shared/_global_config.py

```diff
@@ -99,6 +100,16 @@ def token(self, t: Union[Optional[str], TokenGenerator, Callable[[], Optional[st
         """Sets the Argo Workflows token at a global level so services can use it."""
         self._token = t
 
+    @property
+    def client_certs(self) -> Optional[Tuple[Path, Path]]:
+        return self._client_certs
+
+    @client_certs.setter
+    def client_certs(self, certs: Tuple[Path, Path] | Callable[[], Tuple[Path, Path]]) -> None:
+        if not all(certs) or not isinstance(certs, tuple):
```

Comment:
@flaviuvadan  I am not sure if I should do any additional checking here, what are your thoughts?

Reply:
Not sure we can perform any checks here, actually, since we have a tuple or a callable. I'd make the `property` `client_certs` do something like 
```
if certs is callable
    return self._certs()
return self._certs
```
and not do any checking. Maybe expand the type signature too to `Path | str` and that's it